packages/traceability/issuer-irec-api/src/pods/certification-request/handlers/get-certification-request-by-certificate.handler.ts

```diff
@@ -0,0 +1,44 @@
+import { IQueryHandler, QueryHandler } from '@nestjs/cqrs';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import {
+    Certificate,
+    GetCertificationRequestByCertificateQuery,
+    CertificationRequest
+} from '@energyweb/issuer-api';
+import { IrecCertificationRequest } from '../irec-certification-request.entity';
+import { FullCertificationRequestDTO } from '../full-certification-request.dto';
+
+@QueryHandler(GetCertificationRequestByCertificateQuery)
+export class GetCertificationRequestByCertificateHandler
+    implements IQueryHandler<GetCertificationRequestByCertificateQuery>
+{
+    constructor(
+        @InjectRepository(CertificationRequest)
+        private readonly repository: Repository<CertificationRequest>,
+        @InjectRepository(Certificate)
+        private readonly certificateRepository: Repository<Certificate>,
+        @InjectRepository(IrecCertificationRequest)
+        private readonly irecRepository: Repository<IrecCertificationRequest>
+    ) {}
+
+    async execute({
+        certificateId
+    }: GetCertificationRequestByCertificateQuery): Promise<FullCertificationRequestDTO> {
+        const certificate = await this.certificateRepository.findOne(certificateId);
+        const certificationRequest = await this.repository.findOne({
+            where: {
+                issuedCertificateTokenId: certificate.tokenId
+            }
+        });
+        const irecCertificationRequest = await this.irecRepository.findOne({
+            certificationRequestId: certificationRequest.id
+        });
+
+        return {
+            ...certificationRequest,
+            irecIssueId: irecCertificationRequest.irecIssueId,
+            userId: irecCertificationRequest.userId
+        };
+    }
+}
```

Comment:
No, we can not. Because of nest DI, we have to provide all the dependencies to the constructor of the inherited class. It won't work even if we try to use query or command bus because we override the command/query name

Reply:
Ok, I've done something you mentioned. I had to pass the same arguments as in parent class. Also I had to remove `private` modifier from both classes because it conflicted

Reply:
@JosephBagaric check please

Reply:
Ok, `super()` should be fine for now. Even though we are using the repositories, at least we are not duplicating the logic of the handler.

Reply:
Could you implement the same `super()` logic into other handlers that you extend? Thanks.

Reply:
done